### DISCUSSION TITLE
Android crash Fatal Exception: java.lang.UnsatisfiedLinkError dlopen failed: library "libTrustWalletCore.so" not found

### DISCUSSION THREAD

**Discussion by taras1vovchenko** [trustwallet/wallet-core#3702] [Q&A]
**Describe the bug**
Greetings! I released android app with wallet-core library and started to get a lot of crashes in crashlytics
Fatal Exception: java.lang.UnsatisfiedLinkError
dlopen failed: library "libTrustWalletCore.so" not found
Crash occurs at the beginning of the app in line 
System.loadLibrary("TrustWalletCore")

**To Reproduce**
add line System.loadLibrary("TrustWalletCore") inside of onCreate of Application class

**Expected behavior**
Library is loaded successfully

**Screenshots**
<img width="1163" alt="image" src="https://github.com/trustwallet/wallet-core/assets/126160819/cb9b49fe-2afd-40f3-93e0-9b9430241334">

**Comment by satoshiotomakan** [comment]
Hi @taras1vovchenko, which version of TrustWalletCore do you use? How do you import TrustWalletCore in your app?
Are you able to reproduce the problem locally?
Did you take a look at the examples here: https://github.com/trustwallet/wallet-core/tree/master/android

**Comment by taras1vovchenko** [comment]
Thanks for the response!
I used in production build version 4.0.11
I imported library using github packages 
maven(
            url = uri("https://maven.pkg.github.com/trustwallet/wallet-core")
        ) {
            credentials {
                username = "..."
                password = "..."
            }
        }
        
Everything was set up step-by-step by your guide.  I've tried to launch and test the app on several devices, but I didn't have any issues with the library. I checked your official example and a few other examples on GitHub, and I have almost the same code

**Comment by satoshiotomakan** [comment]
Hi @taras1vovchenko, probably, there is a mistake in android package config.
@AndrewChupin if you have time, could you please advice what could be the problem?

**Comment by tamimattafi** [comment]
Hello @satoshiotomakan,  @AndrewChupin! Any updates on this?
I'm receiving similar crash reports from crashlytics on certain devices using trust core version `4.0.8`
```
Brand:OnePlus
Model:OnePlus8Pro
Version:Android 11
```

We are calling `System.loadLibrary("TrustWalletCore")` lazily during our app's use flow and not in `onCreate` (The app is fully loaded, we call this function when we want to call some trust-core functions_
